aws/aws_kms.go

```diff
@@ -155,16 +158,22 @@ func (a *awsKmsSecrets) GetSecret(
 	if len(cipherBlob) == 0 {
 		cipherBlob = []byte(secretId)
 	}
+	// AWS KMS api requires the cipherBlob to be in base64 decoded format.
+	// Check if it is encoded and decode if required.
+	decodedCipherBlob, err = base64.StdEncoding.DecodeString(string(cipherBlob))
+	if err != nil {
+		decodedCipherBlob = cipherBlob
+	}
 	input := &kms.DecryptInput{
 		EncryptionContext: getAWSKeyContext(keyContext),
-		CiphertextBlob:    cipherBlob,
+		CiphertextBlob:    decodedCipherBlob,
 	}
 	output, err := a.client.Decrypt(input)
 	if err != nil {
 		return nil, err
 	}
 	secretData := make(map[string]interface{})
-	secretData[secretIdKey] = output.Plaintext
+	secretData[secretIdKey] = string(output.Plaintext)
```

Comment:
does the client expect it to be in string now?

Reply:
yes.